homework-submission/todoapp/src/Components/AddForm.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import DatePicker from 'react-datepicker';
+import moment from 'moment';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+class AddTodo extends Component {
+  state = {
+    description: '',
+    deadline: '',
+    done: false,
+    startDate: moment()
+  }
+  handleTodoChange = (e) => {
+    this.setState({
+      [e.target.id]: e.target.value
+    })
+    console.log(this.state)
+  }
+  handleDate = (date) => {
+    this.setState({
+      deadline: date.format("YYYY-MM-DD"),
+      startDate: date
+    })
+
+  }
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.addTodo(this.state)
+    this.setState({
+      description: '',
+      deadline: '',
+      done: false,
+      startDate: moment()
+    })
+
+  }
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handleSubmit}>
+          <div>
+            <label>Add new todo</label>
+            <input type='text' id='description' onChange={this.handleTodoChange} required />
```

Comment:
Oops you forgot to add the `value` attribute 🙃 .

And you are setting the state to be empty on line 30, which will make the next submit click empty item even if it's `required`.